checkers.go

```diff
@@ -90,6 +90,10 @@ func (checker *notChecker) Info() *CheckerInfo {
 func (checker *notChecker) Check(params []interface{}, names []string) (result bool, error string) {
 	result, error = checker.sub.Check(params, names)
 	result = !result
+	if result {
+		// clear error message if the new result is true
+		error = ""
```

Comment:
Is there a test for this?

Reply:
There was only an implicit test for this, I added an explicit one now.